### PR TITLE
Add unsigned long type

### PR DESCRIPTION
**Pull request by grind0x**
This patch adds `unsigned long` type as follows.
  - The `unsigned long` type is an unsigned integer type that has values
    in the range [0, 4294967295].

ISSUE=#158

### CODE REVIEW

File: test/test_interface.idl [hunk 1]
```diff
@@ -32,6 +32,7 @@ interface TestInterface {
   byte byteMethod(byte number);
   octet octetMethod(octet number);
   short shortMethod(short number);
+	unsignedLong unsignedlongMethod(unsignedLong number);
```
Comment (hwanseung) [inline test/test_interface.idl]: nit: adjust indents

**Comment by hwanseung** [comment]
we have to add `ISSUE=#ISSUE_NUM or none` at end of PR message.
refer to already merged https://github.com/lunchclass/bacardi/pull/153

**Comment by hwanseung** [comment]
and please use capital case at the first letter of PR title.

**Review by romandev — COMMENTED** [review]
Also, please update your name and your email in AUTHORS file and include it in this patch.

**Comment by ryonghyun** [comment]
I have modified what you mentioned, can you check whether I did correctly?

File: test/test_interface.h [hunk 2]
```diff
@@ -36,6 +36,7 @@ class TestInterface {
   int8_t ByteMethod(int8_t number);
   uint8_t OctetMethod(uint8_t number);
   short ShortMethod(short number);
+  uint32_t UnsignedlongMethod(uint32_t number);
```
Comment (romandev) [inline test/test_interface.h]: Should be ```UnsignedLong```

File: test/test_interface.idl [hunk 3]
```diff
@@ -32,6 +32,7 @@ interface TestInterface {
   byte byteMethod(byte number);
   octet octetMethod(octet number);
   short shortMethod(short number);
+  unsignedLong unsignedlongMethod(unsignedLong number);
```
Comment (romandev) [inline test/test_interface.idl]: Return type and parameter type should be ```unsinged long```.
Also, method name should be ```unsignedLongMethod```

**Review by romandev — COMMENTED** [review]
Looks good overall
I left minor comments.

**Review by romandev — COMMENTED** [review]
Also, please rebase this patch.

File: test/test_interface.idl (hunk 3 above)
Comment (ryonghyun) [reply test/test_interface.idl]: If I insert a white space between unsigned and long. it causes an error.
I have tried to solve this problem, but I failed.

**Comment by romandev** [comment]
I've checked the issue with your patch. So, this happens due to camelcase filter.
We expect the result of the filter with ```unsigned short``` as follows:
```UnsignedShort```
but actually got
```Unsigned short```

/cc @yjaeseok

**Comment by romandev** [comment]
@usefulhyun FYI, the camelcase(pascalcase) issue was resolved. https://github.com/lunchclass/bacardi/issues/168
Could you please rebase this patch and retry to fix again?

**Comment by romandev** [comment]
@usefulhyun You should run `./bacardi format` to format your changed source code. :)

**Review by yjaeseok — APPROVED** [review]
lgtm
thank you for your contribution

**Comment by ryonghyun** [comment]
Thank you all for your help.

**Pull request merged**